pkg/operator/encryption/key_controller_encryption.go

```diff
@@ -134,22 +134,27 @@ func (c *encryptionKeyController) checkAndCreateKeys() error {
 		keySecret := c.generateKeySecret(gr, nextKeyID)
 		_, createErr := c.secretClient.Create(keySecret)
 		if errors.IsAlreadyExists(createErr) {
-			// TODO maybe drop all of this Get logic
-			actualKeySecret, getErr := c.secretClient.Get(keySecret.Name, metav1.GetOptions{})
-			errs = append(errs, getErr)
-			if getErr == nil {
-				keyGR, _, actualKeyID, validKey := secretToKey(actualKeySecret, c.encryptedGRs)
-				if valid := keyGR == gr && actualKeyID == nextKeyID && validKey; valid {
-					continue // we made this key earlier
-				}
-				// TODO we can just get stuck in degraded here ...
-				errs = append(errs, fmt.Errorf("%s secret %s is in invalid state, new keys cannot be created", gr, keySecret.Name))
-			}
+			errs = append(errs, c.validateExistingKey(keySecret, gr, nextKeyID))
+			continue
 		}
 		errs = append(errs, createErr)
 	}
-	// we do not filter using IsAlreadyExists as the remaining ones are actual errors
-	return utilerrors.FilterOut(utilerrors.NewAggregate(errs), errors.IsNotFound)
+	return utilerrors.NewAggregate(errs)
+}
+
+func (c *encryptionKeyController) validateExistingKey(keySecret *corev1.Secret, gr schema.GroupResource, keyID uint64) error {
+	actualKeySecret, err := c.secretClient.Get(keySecret.Name, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	keyGR, _, actualKeyID, validKey := secretToKey(actualKeySecret, c.encryptedGRs)
+	if valid := keyGR == gr && actualKeyID == keyID && validKey; !valid {
+		// TODO we can just get stuck in degraded here ...
+		return fmt.Errorf("%s secret %s is in invalid state, new keys cannot be created", gr, keySecret.Name)
```

Comment:
At this point, we cannot make progress, should we be more specific and tell a human operator what to do? (e.g. remove the invalid key). Could we also emit an `event`?  Or perhaps we should force a new key?

Reply:
how does it differ from https://github.com/enj/cluster-kube-apiserver-operator/blob/enj/f/enc_config_ref1/pkg/operator/encryption/key_controller_encryption.go#L198? why this is not considered "call support"?